Examples/MAX32655/Unity_Test/test/test_runner.c

```diff
@@ -0,0 +1,46 @@
+/* AUTOGENERATED FILE. DO NOT EDIT. */
+
+/*=======Test Runner Used To Run Each Test Below=====*/
+#define RUN_TEST(TestFunc, TestLineNum)            \
+    {                                              \
+        Unity.CurrentTestName = #TestFunc;         \
+        Unity.CurrentTestLineNumber = TestLineNum; \
+        Unity.NumberOfTests++;                     \
+        if (TEST_PROTECT()) {                      \
+            setUp();                               \
+            TestFunc();                            \
+        }                                          \
+        if (TEST_PROTECT()) {                      \
+            tearDown();                            \
+        }                                          \
+        UnityConcludeTest();                       \
+    }
+
+/*=======Automagically Detected Files To Include=====*/
+#include "unity.h"
+#include <setjmp.h>
+#include <stdio.h>
+
+/*=======External Functions This Runner Calls=====*/
+extern void setUp(void);
+extern void tearDown(void);
+extern void test_simple_add_ok(void);
+extern void test_simple_add_fail(void);
+
+/*=======Test Reset Option=====*/
+void resetTest(void);
+void resetTest(void)
+{
+    tearDown();
+    setUp();
+}
+
+/*=======MAIN=====*/
+int main(void)
```

Comment:
I see this here now...  so should we remove our usual main.c file and just run this instead?

Reply:
The flow of running unity while developing is basically when you run make, it builds the project for the target and runs the unit test on the PC so you can quickly test non hardware dependent features. Of course you can test hardware too on target with it, but typically you mock any platform specific stuff just to test the top level.